deps/rabbit/src/rabbit_fifo_client.erl

```diff
@@ -732,10 +730,22 @@ maybe_auto_ack(false, {deliver, Tag, _Ack, Msgs} = Deliver, State0) ->
     {State, Actions} = settle(Tag, MsgIds, State0),
     {ok, State, [Deliver] ++ Actions}.
 
-
+handle_delivery(_Leader, {delivery, Tag, [_ | _] = IdMsgs},
+                #state{consumer_deliveries = CDels0} = State0) when not is_map_key(Tag, CDels0) ->
+    %% Note:
+    %% https://github.com/rabbitmq/rabbitmq-server/issues/3729
+    %%
+    %% Previously there had been a default value in the consumer deliveries map
+    %% (old code: maps:get(Tag, CDels0, #consumer{last_msg_id = -1}))
+    %% to cover cases where the consumer had been cancelled. We can't restore
```

Comment:
Perhaps we don't need the reference to how it used to work in the bad old days. :)

Let just explain why we're returning here.

Reply:
haha OK